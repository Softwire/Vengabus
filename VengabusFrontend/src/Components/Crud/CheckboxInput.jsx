import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {string} text The name of the property that is being set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;
        const leftDivStyle = css`
            display: inline-block;
            vertical-align: middle;
            min-width: 150px;
        `;
        const rightDivStyle = css`
            display: inline-block;
            vertical-align: middle;
            padding-top: 2px;
        `;

        return (
            <div>
                <div className={leftDivStyle} >
                    {
                        this.props.tooltip ? (
                            <span className={leftAlign}>
                                {this.props.text}
                                <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                    <Glyphicon glyph="info-sign" className={glyphStyle} />
                                </OverlayTrigger>
                            </span>
                        ) : (
                                <span className={leftAlign}>{this.props.text}</span>
                            )
                    }
                </div>
                <div className={rightDivStyle} >
                    <Checkbox
                        className={leftAlign}
                        checked={this.props.data}
                        onChange={(event) => {
                            this.props.onChange(event.target.checked);
                        }}
                    />
                </div>
            </div>
        );
    }

}


