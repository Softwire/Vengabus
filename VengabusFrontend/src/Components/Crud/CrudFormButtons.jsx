import React, { Component } from 'react';
import { css } from 'emotion';
import { ButtonGroup } from 'react-bootstrap';
import { ButtonWithConfirmationModal } from '../ButtonWithConfirmationModal';

/**
 * @prop {string} endpointType The type of endpoint we are editing. Use EndpointTypes in Helpers.
 * @prop {string} selectedEndpoint Name of the selected endpoint.
 * @prop {boolean} buttonsDisabled  If true then these buttons are disabled.
 * @prop {function} updateEndpoint Function that updates the endpoint.
 * @prop {function} resetFields Function that resets the input fields to their original value when called.
 */
export class CrudFormButtons extends Component {

    render() {
        const buttonFormStyle = css`
            text-align: center;
            padding-bottom: 15px;
        `;

        return (
            <form className={buttonFormStyle}>
                <ButtonGroup>
                    <ButtonWithConfirmationModal
                        id="updateButton"
                        buttonText={"Update"}
                        buttonStyle="default"
                        modalInternalStyle="info"
                        buttonDisabled={this.props.buttonsDisabled}
                        modalTitle={"Update " + this.props.endpointType}
                        modalBody={
                            <React.Fragment>
                                <p>{"Following " + this.props.endpointType + " will be updated: " + this.props.selectedEndpoint}</p>
                                <p>{"Confirm action?"}</p>
                            </React.Fragment>
                        }
                        confirmButtonText={"Update"}
                        confirmAction={this.props.updateEndpoint}
                    />
                    <ButtonWithConfirmationModal
                        id="resetButton"
                        buttonText={"Reset Fields"}
                        buttonDisabled={this.props.buttonsDisabled}
                        modalInternalStyle="warning"
                        modalTitle={"Reset all fields"}
                        modalBody={
                            <React.Fragment>
                                <p>Are you sure you want to reset ALL fields of the current {this.props.endpointType}?</p>
                                <p>Note: resetting the fields here will have NO effect on the {this.props.endpointType} in Azure.</p>
                            </React.Fragment>
                        }
                        confirmButtonText={"Reset"}
                        confirmAction={this.props.resetFields}
                    />
                </ButtonGroup>
            </form>
        );
    }
}