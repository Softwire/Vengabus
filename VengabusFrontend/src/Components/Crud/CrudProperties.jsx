import React, { Component } from 'react';
import { css } from 'emotion';
import classNames from 'classnames';
import { DataTable } from '../DataTable';
import { PropertyInput } from './PropertyInput';
import { TimeSpanInput } from './TimeSpanInput';
import { InputLabel } from './InputLabel';

/**
 * @prop {string} endpointType The type of endpoint we are editing. Use EndpointTypes in Helpers.
 * @prop {object} newEndpointData The edited description of the endpoint.
 * @prop {function} getEditableAndReadOnlyProperties Gets:
     *  {string[]} Property names for editable properties.
     *  {object} Display name and display value pairs for read-only properties.
 * @prop {function} handlePropertyChange Function that is called when a property is changed in the form.
 */
export class CrudProperties extends Component {

    /**
     * @returns {string} Class name for standard hr component style used in this interface.
     */
    getHrStyle = () => {
        return css`
            color: black;
            background-color: black;
            height: 1px;
            width: 98%;
        `;
    }

    /**
     * @returns {Object <string, string>} Maps from property name to text to be displayed on info hover over.
     */
    getTooltips = () => {
        return {
            requiresSession:
                `True if the receiver application can only receive from the ${this.props.endpointType} through a MessageSession; false if a ${this.props.endpointType} cannot receive using MessageSession.`,
            autoDeleteOnIdle:
                `The idle time span after which the ${this.props.endpointType} is automatically deleted. The minimum duration is 5 minutes.`,
            maxDeliveryCount:
                'A message is automatically deadlettered after this number of deliveries.',
            enableDeadletteringOnMessageExpiration:
                `Sets whether this ${this.props.endpointType} has dead letter support when a message expires.`
        };
    }

    /**
     * @returns {Object <string, {label: string, value: any}>[]} Maps from property name to permitted dropdown options where required.
     */
    getDropdownOptions = () => {
        const statusOptions = [{ label: 'Active', value: 'Active' }, { label: 'Disabled', value: 'Disabled' }];
        return {
            status: statusOptions,
            topicStatus: statusOptions,
            subscriptionStatus: statusOptions
        };
    }

    /**
     * @returns {Object <string, class>} Maps property name to component type for the cases where this is required.
     */
    getObjectPropertyToComponent = () => {
        return {
            autoDeleteOnIdle: TimeSpanInput
        };
    }

    /**
     * @returns {node[]} Input components for editable properties.
     * @returns {node} Data table for read-only properties.
     */
    getEditableAndReadOnlyPropertyComponents = () => {
        const [editableProperties, readOnlyProperties] = this.props.getEditableAndReadOnlyProperties();
        const tableStyle = css`
            width: 98%;
            padding-left: 20px;
        `;
        const rowStyle = css`
            text-align: left;
        `;
        const colProps = [{ dataField: 'name', text: 'Property Name', headerStyle: { textAlign: 'left' } }, { dataField: 'value', headerStyle: { textAlign: 'left' } }];
        const editablePropertyInputs = this.getEditablePropertyInputs(editableProperties);
        const readOnlyPropertyTable =
            <div>
                <hr className={this.getHrStyle()} />
                <div className={tableStyle} >
                    <DataTable
                        dataToDisplay={readOnlyProperties}
                        uniqueKeyColumn='name'
                        colProps={colProps}
                        rowClasses={rowStyle}
                        bordered={false}
                    />
                </div>
                <hr className={this.getHrStyle()} />
            </div>;
        return [editablePropertyInputs, readOnlyPropertyTable];
    }

    /**
     * @param {string[]} editableProperties Property names for editable properties.
     * @returns {node[]} Array of jsx elements for property inputs.
     */
    getEditablePropertyInputs = (editableProperties) => {
        let editablePropertyInputs = [];
        editablePropertyInputs.push(<hr className={this.getHrStyle()} key={0} />);
        for (let i = 0; i < editableProperties.length; i++) {
            const property = editableProperties[i];
            editablePropertyInputs.push(
                <div key={`input${i}`}>
                    <InputLabel
                        propertyName={property.charAt(0).toUpperCase() + property.substr(1)}
                        tooltipText={this.getTooltips()[property]}
                    />
                    <PropertyInput
                        inputData={this.props.newEndpointData[property]}
                        onChange={(data) => this.props.handlePropertyChange(data, property)}
                        complexInputComponentType={this.getObjectPropertyToComponent()[property]}
                        options={this.getDropdownOptions()[property]}
                    />
                </div>
            );
        }
        editablePropertyInputs.push(<hr className={this.getHrStyle()} key={1} />);
        return editablePropertyInputs;
    }

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const headerStyle = css`
            padding-top: 10px;
            font-weight: bold;
            font-size: 1.6em;
        `;
        let [editablePropertyInputs, readOnlyPropertyTable] = this.getEditableAndReadOnlyPropertyComponents();

        return (
            <div>
                <p className={classNames(leftAlign, headerStyle)}>Read-Only Properties</p>
                {readOnlyPropertyTable}

                <p className={classNames(leftAlign, headerStyle)}>Editable Properties</p>
                {editablePropertyInputs}
            </div>
        );
    }
}