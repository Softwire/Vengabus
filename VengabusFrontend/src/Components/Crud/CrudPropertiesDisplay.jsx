import React, { Component } from 'react';
import { css } from 'emotion';
import classNames from 'classnames';
import { DataTable } from '../DataTable';
import { PropertyInput } from './PropertyInput';
import { InputLabel } from './InputLabel';

/**
 * @prop {string} endpointType The type of endpoint we are editing. Use EndpointTypes in Helpers.
 * @prop {object} newEndpointData The edited description of the endpoint.
 * @prop {object} endpointProperties And object with 'editable', 'setAtCreation' & 'readonly' properties, each of which is an Array of propertyConfig objects.
 * @prop {function} handlePropertyChange Function that is called when a property is changed in the form.
 */
export class CrudPropertiesDisplay extends Component {

    /**
     * @returns {string} Class name for standard hr component style used in this interface.
     */

    /**
     * @returns {node} Data table for read-only properties.
     */
    getReadOnlyPropertyTable = () => {
        const allProperties = this.props.endpointProperties;
        const readOnlyProperties = [...allProperties.readonly, ...allProperties.setAtCreation];
        const propsWithValues = readOnlyProperties.map(prop => {
            return {
                name: prop.displayLabel,
                value: this.props.newEndpointData[prop.propertyName] // qqMDM mostRecentDeadletter will start as null. Look for this and replace with spinner?
            };
        });
        const colProps = [
            { dataField: 'name', text: 'Property Name', width: 20, headerStyle: { textAlign: 'left' } },
            { dataField: 'value', width: 80, headerStyle: { textAlign: 'left' } }
        ];

        const tableStyle = css`
            width: 98%;
            padding-left: 20px;
        `;
        const rowStyle = css`
            text-align: left;
        `;

        return (
            <div className={tableStyle} >
                <DataTable
                    dataToDisplay={propsWithValues}
                    uniqueKeyColumn='name'
                    colProps={colProps}
                    rowClasses={rowStyle}
                    bordered={false}
                />
            </div>
        );
    }

    /**
     * @param {string[]} editableProperties Property names for editable properties.
     * @returns {node[]} Array of jsx elements for property inputs.
     */
    getEditablePropertyInputs = () => {
        const editableProperties = this.props.endpointProperties.editable;

        const editablePropertyInputs = [];
        for (let i = 0; i < editableProperties.length; i++) {
            const propertyConfig = editableProperties[i];
            const property = propertyConfig.propertyName;
            editablePropertyInputs.push(
                <div key={`input${i}`}>
                    <InputLabel
                        propertyName={propertyConfig.displayLabel}
                        tooltipText={propertyConfig.tooltipText}
                    />
                    <PropertyInput
                        propertyName={property}
                        inputData={this.props.newEndpointData[property]}
                        onChange={(data) => this.props.handlePropertyChange(data, property)}
                        complexInputComponentType={propertyConfig.component}
                        options={propertyConfig.dropdownValues}
                    />
                </div>
            );
        }
        return editablePropertyInputs;
    }

    render() {
        const hrStyle = css`
            color: black;
            background-color: black;
            height: 1px;
            width: 98%;
        `;

        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const headerStyle = css`
            padding-top: 10px;
            font-weight: bold;
            font-size: 1.6em;
        `;

        const readOnlyPropertyTable = this.getReadOnlyPropertyTable();
        const editablePropertyInputs = this.getEditablePropertyInputs();
        const styledHr = (<hr className={hrStyle} />);
        return (
            <div>
                <p className={classNames(leftAlign, headerStyle)}>Read-Only Properties</p>
                {styledHr}
                {readOnlyPropertyTable}
                {styledHr}

                <p className={classNames(leftAlign, headerStyle)}>Editable Properties</p>
                {styledHr}
                {editablePropertyInputs}
                {styledHr}
            </div>
        );
    }
}