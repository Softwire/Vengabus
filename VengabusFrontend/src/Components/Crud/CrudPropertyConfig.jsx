import { EndpointTypes } from "../../Helpers/EndpointTypes";
import { TimeSpanInput } from "./TimeSpanInput";

export const getQueueCrudProperties = () => {
    const queue = EndpointTypes.QUEUE;
    return {
        readonly: [
            new PropertyConfig('activeMessageCount'),
            new PropertyConfig('deadletterMessageCount'),
            new PropertyConfig('mostRecentDeadletter'),
        ],
        setAtCreation: [
            new PropertyConfig('enablePartitioning'),
            new PropertyConfig('requiresSession').WithCommonTooltip(queue),
            new PropertyConfig('requiresDuplicateDetection'),
        ],
        editable: [
            new PropertyConfig('supportOrdering'),
            new PropertyConfig('autoDeleteOnIdle').WithCommonTooltip(queue).WithCustomInput(TimeSpanInput),
            new PropertyConfig('enableDeadletteringOnMessageExpiration').WithCommonTooltip(queue),
            new PropertyConfig('maxDeliveryCount').WithCommonTooltip(queue),
            new PropertyConfig('maxSizeInMegabytes'),
            new PropertyConfig('status').WithDropdown(getStatusDropdownOptions()),
        ]
    };
};

export const getSubscriptionCrudProperties = () => {
    const sub = EndpointTypes.SUBSCRIPTION;
    return {
        readonly: [
            new PropertyConfig('topicName'),
            new PropertyConfig('activeMessageCount'),
            new PropertyConfig('deadletterMessageCount'),
            new PropertyConfig('mostRecentDeadletter'),
        ],
        setAtCreation: [
            new PropertyConfig('requiresSession').WithCommonTooltip(sub),
        ],
        editable: [
            new PropertyConfig('autoDeleteOnIdle').WithCommonTooltip(sub).WithCustomInput(TimeSpanInput),
            new PropertyConfig('enableDeadletteringOnMessageExpiration').WithCommonTooltip(sub),
            new PropertyConfig('maxDeliveryCount').WithCommonTooltip(sub),
            new PropertyConfig('subscriptionStatus').WithDropdown(getStatusDropdownOptions()),
        ]
    };
};

export const getTopicCrudProperties = () => {
    const topic = EndpointTypes.TOPIC;
    return {
        readonly: [
            new PropertyConfig('subscriptionCount'),
        ],
        setAtCreation: [
            new PropertyConfig('requiresDuplicateDetection'),
            new PropertyConfig('enablePartitioning'),
        ],
        editable: [
            new PropertyConfig('supportOrdering'),
            new PropertyConfig('autoDeleteOnIdle').WithCommonTooltip(topic).WithCustomInput(TimeSpanInput),
            new PropertyConfig('maxSizeInMegabytes'),
            new PropertyConfig('topicStatus').WithDropdown(getStatusDropdownOptions()),
        ]
    };
};

class PropertyConfig {
    constructor(propertyName, displayLabel) {
        this.propertyName = propertyName;
        this.displayLabel = displayLabel || propertyName.charAt(0).toUpperCase() + propertyName.substr(1);
    }

    WithCommonTooltip = (endpointType) => {
        this.tooltipText = commonTooltipText(this.propertyName, endpointType);
        return this;
    }

    WithCustomTooltip = (tooltipText) => {
        this.tooltipText = tooltipText;
        return this;
    }

    WithDropdown = (dropdownOptions) => {
        this.dropdownValues = dropdownOptions;
        return this;
    }

    WithCustomInput = (inputComponent) => {
        this.component = inputComponent;
        return this;
    }
}

function commonTooltipText(field, endpointType) {
    const texts = {
        requiresSession:
            `True if the receiver application can only receive from the ${endpointType} through a MessageSession; false if a ${endpointType} cannot receive using MessageSession.`,
        autoDeleteOnIdle:
            `The idle time span after which the ${endpointType} is automatically deleted. The minimum duration is 5 minutes.`,
        maxDeliveryCount:
            'A message is automatically deadlettered after this number of deliveries.',
        enableDeadletteringOnMessageExpiration:
            `Sets whether this ${endpointType} has dead letter support when a message expires.`
    };
    return texts[field];
}

/**
 * @returns {Object <string, {label: string, value: any}>[]} Object for dropdown of possible EndpointStatuses.
 */
function getStatusDropdownOptions() {
    return getDropdownOptions(['Active', 'Disabled']);
}

/**
 * @param {string[]} textValues Possible strings to select.
 * @returns {Object <string, {label: string, value: any}>[]} Maps from property name to permitted dropdown options where required.
 */
function getDropdownOptions(textValues) {
    return textValues.map(text => { return { label: text, value: text }; });
}
