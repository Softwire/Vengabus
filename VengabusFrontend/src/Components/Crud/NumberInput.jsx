import React, { Component } from 'react';
import { css } from 'emotion';
import { FormGroup, FormControl } from "react-bootstrap";

/**
 * @prop {number} inputData The data that is set by the input.
 * @prop {function} onChange The function that is called when the value of the input is changed.
 */
export class NumberInput extends Component {

    render() {
        const inputStyle = css`
            width: 100px;
            float: left;
            padding-left: 5px;
        `;
        const formStyle = css`
            padding-left: 10px;
        `;
        const divStyle = css`
            display: inline-block;
            vertical-align: middle;
            padding-top: 15px;
        `;

        return (
            <div className={divStyle} >
                <form className={formStyle}>
                    <FormGroup className={inputStyle}>
                        <FormControl
                            type="number"
                            value={this.props.inputData}
                            placeholder="Enter Number"
                            onChange={(event) => this.props.onChange(parseInt(event.target.value, 10))}
                        />
                    </FormGroup>
                </form>
            </div>
        );
    }
}