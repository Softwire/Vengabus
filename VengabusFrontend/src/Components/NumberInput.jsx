import React, { Component } from 'react';
import { css } from 'emotion';
import {
    FormGroup,
    FormControl,
    OverlayTrigger,
    Glyphicon
} from "react-bootstrap";

/**
 * @prop {number} data The data that is set by the input.
 * @prop {string} text The name of the property that is being set by the input.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the input is changed.
 */
export class NumberInput extends Component {

    render() {
        const inputStyle = css`
            width: 14%;
            float: left;
            padding-left: 5px;
        `;
        const formStyle = css`
            padding-left: 10px;
        `;
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <form className={formStyle}>
                    <FormGroup className={inputStyle}>
                        <FormControl
                            type="number"
                            value={this.props.data}
                            placeholder="Enter Number"
                            onChange={(event) => this.props.onChange(parseInt(event.target.value, 10))}
                        />
                    </FormGroup>
                </form>
            </div>
        );
    }
}