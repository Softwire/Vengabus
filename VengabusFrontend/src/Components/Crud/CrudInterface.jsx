import React, { Component } from 'react';
import _ from 'lodash';
import { CrudTitle } from './CrudTitle';
import { CrudPropertiesDisplay } from './CrudPropertiesDisplay';
import { CrudFormButtons } from './CrudFormButtons';

/**
 * @prop {string} endpointType The type of endpoint we are editing. Use EndpointTypes in Helpers.
 * @prop {string} selectedEndpoint Name of the selected endpoint.
 * @prop {string} parentTopic The parent topic of the subscription being edited. Only required for subscriptions.
 * @prop {object} originalEndpointData The original description of the endpoint.
 * @prop {object} newEndpointData The edited description of the endpoint.
 * @prop {object} endpointProperties And object with 'editable', 'setAtCreation' & 'readonly' properties, each of which is an Array of propertyConfig objects.
 * @prop {function} handlePropertyChange Function that is called when a property is changed in the form.
 * @prop {function} renameEndpoint  Function that renames the endpoint when called.
 * @prop {function} updateEndpoint Function that updates the endpoint using the data stored in newEndpointData.
 * @prop {function} deleteEndpoint Function that deletes the selected endpoint when called.
 * @prop {function} resetFields Function that resets the input fields to their original value when called.
 */
export class CrudInterface extends Component {

    render() {
        return (
            <div>
                <br />
                <CrudTitle
                    endpointType={this.props.endpointType}
                    selectedEndpoint={this.props.selectedEndpoint}
                    parentTopic={this.props.parentTopic}
                    renameDisabled={this.props.originalEndpointData.enablePartitioning}
                    renameEndpoint={this.props.renameEndpoint}
                    deleteEndpoint={this.props.deleteEndpoint}
                />
                <CrudPropertiesDisplay
                    endpointType={this.props.endpointType}
                    newEndpointData={this.props.newEndpointData}
                    endpointProperties={this.props.endpointProperties}
                    handlePropertyChange={this.props.handlePropertyChange}
                />
                <CrudFormButtons
                    endpointType={this.props.endpointType}
                    selectedEndpoint={this.props.selectedEndpoint}
                    buttonsDisabled={_.isEqual(this.props.originalEndpointData, this.props.newEndpointData)}
                    updateEndpoint={this.props.updateEndpoint}
                    resetFields={this.props.resetFields}
                />
            </div>
        );

    }
}
