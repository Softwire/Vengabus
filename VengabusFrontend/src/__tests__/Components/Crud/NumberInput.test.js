import React from 'react';
import { NumberInput } from '../../../Components/Crud/NumberInput';
import renderer from 'react-test-renderer';
import { Tooltip } from 'react-bootstrap';

it('renders correctly with given props', () => {
    const tooltipText = 'Test tooltip.';
    let checkboxInput = renderer.create(
        <NumberInput
            inputData={42}
            text="numeric data"
            onChange={() => { }}
            tooltipText={tooltipText}
        />
    );
    expect(checkboxInput.toJSON()).toMatchSnapshot();
});