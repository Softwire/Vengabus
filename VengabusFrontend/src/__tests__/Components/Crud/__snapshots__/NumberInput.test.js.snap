// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with given props 1`] = `
.emotion-0 {
  width: 100px;
  float: left;
  padding-left: 5px;
}

.emotion-1 {
  padding-left: 10px;
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  padding-top: 15px;
}

<div
  className="emotion-2"
>
  <form
    className="emotion-1"
  >
    <div
      className="emotion-0 form-group"
    >
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Enter Number"
        type="number"
        value={42}
      />
    </div>
  </form>
</div>
`;
